File: ai-service/app/models/business_data.py
from sqlalchemy import Column, BigInteger,Integer, Float, String, Date
from app.database import Base

class BusinessData(Base):
    __tablename__ = "business_data"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(BigInteger, nullable=False)
    dataset_id = Column(BigInteger, nullable=False)

    date = Column(Date, nullable=False)
    product = Column(String, nullable=False)
    category = Column(String, nullable=False)
    quantity = Column(Integer, nullable=False)
    price = Column(Float, nullable=False)
    

File: laravel-backend/app/Models/BusinessData.php
<?php
namespace App\Models;
use Illuminate\Database\Eloquent\Model;

class BusinessData extends Model
{
    protected $table = 'business_data';

    public function dataset()
    {
        return $this->belongsTo(Dataset::class, 'dataset_id');
    }

    protected $fillable = [
        'dataset_id',
        'date',
        'month',
        'year',
        'product',
        'category',
        'quantity',
        'price',
        'total',
    ];
}
?>